src/ChaiVM/interpreter/code-manager/code-manager.cpp

```diff
@@ -0,0 +1,141 @@
+#include <bit>
+#include <cassert>
+
+#include "ChaiVM/interpreter/code-manager/code-manager.hpp"
+
+namespace chai::interpreter {
+
+void CodeManager::load(const std::filesystem::path &path) {
+    std::ifstream input_file(path, std::ios::binary | std::ios::in);
+    if (input_file.good() && input_file.is_open()) {
+        loadPool(input_file);
+        Immidiate func_count = 0;
+        input_file.read(reinterpret_cast<char *>(&func_count),
+                        sizeof func_count);
+        funcs_ = std::vector<Function>{};
+        for (int i = 0; i < func_count; ++i) {
+            loadFunction(input_file);
+        }
+        input_file.close();
+    } else {
+        throw std::invalid_argument(std::string{"Invalid path "} +
+                                    path.string());
+    }
+}
+
+void CodeManager::loadPool(std::istream &istream) {
+    if (!istream.good()) {
+        throw std::invalid_argument(std::string{"Bad input stream"});
+    }
+    Immidiate constants = 0;
+    istream.read(reinterpret_cast<char *>(&constants), sizeof constants);
+    dispatch_ = std::vector<Immidiate>(constants, -1);
```

Comment:
@c71n93 constants is size of the vector, -1 is like poison. For example, to avoid cases when the false test passes because there is a lucky value there.

Reply:
@levBagryansky I think here you better to do `dispatch_.resize(constants)`
1. This removes the use of an unnecessary move = operator
2. Is it right to fill in a `dispatch_` with `-1` to fit tests?

Reply:
@c71n93 the code is just safer. this is my motivation